File: android/app/src/main/java/com/moonshrd/models/MessageModel.java
package com.moonshrd.models;

import com.google.gson.annotations.SerializedName;

public class MessageModel {
    @SerializedName("Id") public String id;
    @SerializedName("Body") public String body;
    @SerializedName("From") public String from;
    @SerializedName("Topic") public String topic;
    @SerializedName("Timestamp") public long timestamp;
    @SerializedName("User") public UserModel user;

    public MessageModel() { }

    public MessageModel(String id, String body, String from, String topic, long timestamp, UserModel user) {
        this.id = id;
        this.body = body;
        this.from = from;
        this.topic = topic;
        this.timestamp = timestamp;
        this.user = user;
    }
}

File: android/app/src/main/java/com/moonshrd/models/MessageModel.kt
package com.moonshrd.models

import com.google.gson.annotations.SerializedName

data class MessageModel(
        @SerializedName("Id")
        var id: String,
        @SerializedName("body")
        var body: String,
        @SerializedName("fromMatrixID")
        var from: String,
        @SerializedName("topic")
        var topic: String,
        @SerializedName("Timestamp")
        var timestamp: Long = 0,
        @SerializedName("User")
        var user: UserModel?,
        @SerializedName("fromPeerID")
        var fromPeerID: String?=null
)



